### PR TITLE
Fix getCurrent API return type

### PR DESCRIPTION
**Pull request by sebasjm**
The documentation does not match the API

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.chrome.com/docs/extensions/reference/api/tabs#method-getCurrent
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by nicolas377 — APPROVED** [review]
Hey @sebasjm, looks good! It might be a bit of a headache to get this merged without any test changes, specifically it might just take a couple of days longer to get a DT maintainer to approve the change. I'll sign off on this, but if you could just change a test somewhere, that would be awesome. If you're ok with waiting for a DT maintainer, then go for it, but if you'd like to add some tests, then if there's no tests for this method, add them, if there is, just update an ExpectType declaration. Again, thanks for your contribution!

**Comment by sebasjm** [comment]
@nicolas377  ACK, I will check the test to see where this can fit.

@matcarlson  from the docs (which is currently in the issue description)

> Gets the tab that this script call is being made from. Returns undefined if called from a non-tab context (for example, a background page or popup view).

The return type doesn't match what the documentation describes or runtime behavior.
The function CAN return `undefined` (which I don't think is nice but we can't change that, right?) and the type declaration here is not open to that possibility. 
The result is worst than wrong since we are using typescript in order to not to have this kind of surprises :)

**Comment by nicolas377** [comment]
Wait, I must have missed something my first time around. My reading now seems to imply that the undefined case ends with the Promise resolving with `undefined`, not the function returning `undefined`. Is there runtime code that sees this happen? I'm gonna retract my approval for now, since I think this change is just a misreading of the docs.

PS: for a bit more evidence, see the "Returns" section of the function. It says `Promise<Tab | undefined>`, not `Promise<Tab | undefined> | undefined`, a critical difference especially if using `.then()`, since js would throw `Cannot read properties of undefined`, and tsc warns because of that. See [this playground](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEBLAdgFxDAZlMCAqUARvAN4BQ88ADjAPZUBc8AzqjCgOYDcZAvmaEiwEmAK7IwqRLWQtEAWyoQAngAU68xMxAAKAJRN1tTdoA8BYgB9440JhQhgAPh6DoceGIlSZ1DVpAAeRgAVWQ7B2B9Q38zC3hrWxB7ZEcnBJtw5MieMgB6PPhUAAsteAB3WhgAa2YyZgUlNVjdPQA6EpBkHQA3KAh4AF50vogAfjaaej1cgqLS5nhgWhBmZAByVDkJBFRmMHmUWvhlWlF4MDOIYHgum9EqIphlTiLaeBxcZkWdJJTHdqdZBkKYmIKhLL-KKA4pdXr9IYjfoTKZUPRkIA) for an example.

**Review by nicolas377 — COMMENTED** [review]
Retracting the approval for now.

**Review by nicolas377 — CHANGES_REQUESTED** [review]
Sorry, forgot github reviews don't work that way.

**Comment by sebasjm** [comment]
>  Is there runtime code that sees this happen?

Yes! that's why this is important. The documentation in chrome say it works like that, please read the docs.

> PS: for a bit more evidence, see the "Returns" section of the function. It says Promise<Tab | undefined>, not Promise<Tab | undefined> | undefined, a critical difference especially if using .then(), since js would throw Cannot read properties of undefined, and tsc warns because of that.

I know exactly what this implies, but typechecks are for runtime safety and if the runtime returns `undefined` then being correct is strictly better than having a nice API.

My current workaround for this is having a wapper function which correctly returns `Promise<Tab | undefined> | undefined`, then checking if the return value is not undeinfed and then `await` for that. Weird? true, but that's the underlying behavior.

Here, in this case, the types brings more confusion instead of helping.

**Comment by nicolas377** [comment]
It could just be me, but it seems as if you're misreading the docs. To be fair, the use of the word "returns" definitely isn't helpful here since it implies that the function would _return_ `undefined` instead of a Promise. However, the provided return type doesn't reflect this behavior, causing me to think that the phrase "returns `undefined`" is referring to resolving the Promise with the value `undefined`. I could very well be wrong however, in which case Chrome needs to update their documentation to reflect this. I would encourage you to prove me wrong, though. I will happily approve this PR if someone can demonstrate .getCurrent() actually _returning_ `undefined` instead of a Promise that resolves to `undefined`.

**Pull request closed without merging**
